earth_enterprise/tutorial/SConstruct

```diff
@@ -62,10 +62,35 @@ def makeSoftLinks(dest_dir, src_dir, excluded_list):
             commands.getstatusoutput('ln -s "' + src_filepath + '" "' +
                                     dest_filepath + '"')
 
+def copyFiles(dest_dir, src_dir, excluded_list):
+    """Make copies of all files in src_dir and place
+    them in dest_dir. Don't copy files in excluded_list.
+    Recurse through all subdirectories of src_dir.
+    """
+    # Strip trailing slashes because they cause trouble later
+    dest_dir = dest_dir.rstrip(os.sep)
+    src_dir = src_dir.rstrip(os.sep)
+    src_contents = os.listdir(src_dir)
+    # Ensure the destination directory exists at each step
+    commands.getstatusoutput("mkdir -p " + dest_dir)
+    for file in src_contents:
+        if file in excluded_list:
+            continue
+        # Use full paths to all files so `ln -s ... ` succeeds
+        dest_filepath = os.path.join(dest_dir, file)
+        src_filepath = os.path.join(src_dir, file)
+        if os.path.isdir(src_filepath):
+            # Recurse for next-lower directory
+            makeSoftLinks(dest_filepath, src_filepath, excluded_list)
+        else:
+            commands.getstatusoutput('cp "' + src_filepath + '" "' +
+                                    dest_filepath + '"')
+
 # Clean install dir (note: the current sconscripts leave stuff behind, like
 # directories).
 if env.GetOption('clean'):
     # clean the search and getomcat install folders.
     commands.getstatusoutput("rm -rf " + tutorials_dir)
 else:
-    makeSoftLinks(root_dir, tutorial_src_dir, ['.svn'])
+    copyFiles(root_dir, tutorial_src_dir, ['.svn'])
```

Comment:
I am guessing historically geep was originally hosted in svn rather than git.  But do we have .svn directories today??  If not, and we don't need the exclude list anymore, could we simplify this to using a cp -R or maybe an existing python function that does recursive copying?

Reply:
Sure. We can certainly refactor it. I wrote it this way to be in line with other scons stuff.